### DISCUSSION TITLE
[Question] Component rendered when use appfile to create application

### DISCUSSION THREAD

**Discussion by LeoLiuYan** [kubevela/kubevela#814] [Q&A]
**Is your feature request related to a problem? Please describe.**

1.  When we use appfile to create a application, the component of application will rendered which mean `spec.parameters` is nil, and then oam runtime create a workload for the component, the workload object's `spec`  is the same as component object's `spec.workload` .  Should component object keep parameters which derived from cue template? 
2.  Should component's `labels` and `annotation` pass to workload object in oam runtime?  Related code may be here: 
     https://github.com/oam-dev/kubevela/blob/master/pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go#L157

**Describe the solution you'd like**
1.  Use cue template parameters as component object's `spec.parameters` when create component.
2. Pass component's `labels` and `annotation` to workload.

**Describe alternatives you've considered**

**Additional context**

**Comment by resouer** [comment]
> Use cue template parameters as component object's spec.parameters when create component.

@LeoLiuYan The intention is using cue template parameters to deprecate Component parameters. We will try to avoid implementing same feature in two places. This will also leave Component + ApplicationConfiguration as pure k8s object carrier and orchestrator (i.e. control object apply order and data input/output). Does this make sense?

I thought the labels are already propagated. /cc @wonderflow

**Comment by wonderflow** [comment]
For 1, like @resouer said, we're trying to deprecate parameters defined in Component. There's no mean using parameter in component when we already have values by creating a component instance.  We're also building [`Application` CRD Object](https://github.com/oam-dev/kubevela/blob/master/apis/core.oam.dev/v1alpha2/application_types.go#L53) which will be the serverside object of appfile(partially without `image build` or any thing that should just be in client side).

For 2, can you describe in which case you want to paas component label and annotation? AFAIK, You can directly put the label and annotation in `spec.workload` of component.

**Comment by LeoLiuYan** [comment]
Got it.  I will look at the implementation of the `Application`.

For question two, when create a component, for example we add `componentName: Cache` to it labels, we wish this label pass to `workoad` object.  Now we could add this label direct to `workload` object, but I think this should do by `OAM runtime`.

Thank you for your reply.   @resouer @wonderflow
